src/components/TodoItem.tsx

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import className from 'classnames';
+import { FormEvent, useState } from 'react';
+import { Todo } from '../types/Todo';
+import * as postService from '../api/todos';
+import { Error } from '../types/Error';
+
+type Props = {
+  todo: Todo,
+  onDeleteTodo: (todoId: number) => void,
+  loadingIds: number[],
+  setTodos: (value: Todo[]) => void,
+  setLoadingIds: React.Dispatch<React.SetStateAction<number[]>>,
+  todos: Todo[]
+};
+
+const USER_ID = 11707;
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDeleteTodo,
+  loadingIds,
+  setTodos,
+  setLoadingIds,
+  todos,
+}) => {
+  const [, setErrorMessage] = useState(Error.None);
+  const [doubleClicked, setDoubleClicked] = useState(false);
+  const [editingTitle, setEditingTitle] = useState(todo.title);
```

Comment:
Consider will be better if you add a generic types everywhere for state for better readability